### PR TITLE
Propose KIP-162 Priority Fee Mechanism

### PR DESCRIPTION
**Pull request by blukat29**
KIP-162 describes a priority fee mechanism for Klaytn transaction type 2 (EthereumDynamicFee).

Discussions to #161

### CODE REVIEW

File: KIPs/kip-162.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+### Transaction ordering
+
+Block transaction ordering depends on client implementation as the ordering is not checked by block validators. However, under the KIP-82 reward scheme, a proposer should prioritize high priority fee transactions for the maximum proposer reward.
+
+Block proposers are recommended to order transactions with descending effective gas price, and the same effective gas price are sorted by transaction arrival time. This discourages the transaction spamming where a sender sends many transactions hoping that at least one is included in the block. Instead, the sender would pay a higher priority fee to ensure the transaction be included in the block.
+
+### Txpool management
+
+A transaction pool should not accept underpriced transactions. A transaction is underpriced if its `tx.gasPrice` or `tx.maxFeePerGas` field is lower than the baseFeePerGas. The transaction pool can use last block's baseFeePerGas or the baseFeePerGas calculated for next block.
```
Comment (ian0371) [inline KIPs/kip-162.md]: > use last block's baseFeePerGas or the baseFeePerGas calculated for next block

Which is it in Klaytn? and could you add this for clarification?

File: KIPs/kip-162.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  - `rewardPercentiles` - (optional) A monotonically increasing list of percentile (between 0 and 100) values. For each block in the requested range, the transactions will be sorted in ascending order by effective tip per gas and sampled at the specified percentiles.
+- Result
+  - `oldestBlock` - Lowest number block of returned range.
+  - `baseFeePerGas` - An array of block base fee per gas. This includes the next block after the newest of the returned range, because this value can be derived from the newest block. Zeroes are returned for pre-Magma blocks.
+  - `gasUsedRatio` - An array of block gas used ratios. Measures the network congestion level. These are calculated as the ratio of block gas used and [KIP-71 MAX_BLOCK_GAS_USED_FOR_BASE_FEE](https://github.com/klaytn/kips/blob/main/KIPs/kip-71.md). If the ratio is above 1, then 1 is returned.
+  - `reward` - A 2D array of effective priority fees per gas. `reward[n][i]` is the `rewardPercentiles[i]`-th percentile effective priority fees per gas among the transactions in the block `oldestBlock + n`. Only returned if `rewardPercentiles` is specified.
```
Comment (ian0371) [inline KIPs/kip-162.md]:
```suggestion
  - `reward` - (optional) A 2D array of effective priority fees per gas. `reward[n][i]` is the `rewardPercentiles[i]`-th percentile effective priority fees per gas among the transactions in the block `oldestBlock + n`. Only returned if `rewardPercentiles` is specified.
```

File: KIPs/kip-162.md (hunk 1 above)
Comment (blukat29) [reply KIPs/kip-162.md]: Fixed in 344af4e. It's the calculated baseFeePerGas of the next (pending) block. The `pool.gasPrice`, which holds the required minimum price, is updated (https://github.com/klaytn/klaytn/blob/v1.12.1/blockchain/tx_pool.go#L488) whenever the txpool receives the ChainHeadEvent (https://github.com/klaytn/klaytn/blob/v1.12.1/blockchain/tx_pool.go#L307).

File: KIPs/kip-162.md (hunk 2 above)
Comment (blukat29) [reply KIPs/kip-162.md]: Fixed in https://github.com/klaytn/kips/commit/344af4e6cd4fa25a1104ad845a73d551a7acf7d3

**Review by ian0371 — APPROVED** [review]

**Review by hyeonLewis — APPROVED** [review]

**Pull request merged**